src/Pretzel/Commands/BakeCommand.cs

```diff
@@ -14,16 +14,20 @@ namespace Pretzel.Commands
 {
     [Shared]
     [Export]
-    [CommandArguments(CommandName = BuiltInCommands.Bake)]
-    public sealed class BakeCommandParameters : BakeBaseCommandParameters
+    [CommandArguments]
+    public sealed class BakeCommandArguments : BakeBaseCommandArguments
     {
         [ImportingConstructor]
-        public BakeCommandParameters(IFileSystem fileSystem) : base(fileSystem) { }
+        public BakeCommandArguments(IFileSystem fileSystem) : base(fileSystem) { }
     }
 
     [Shared]
-    [CommandInfo(CommandName = BuiltInCommands.Bake, CommandDescription = "transforming content into a website")]
-    public sealed class BakeCommand : IPretzelCommand
+    [CommandInfo(
+        CommandName = BuiltInCommands.Bake,
+        CommandDescription = "transforming content into a website",
+        CommandArgumentsType = typeof(BakeCommandArguments)
```

Comment:
This information can't be retrieved by the class type generic parameter?

Reply:
Nope, we don't have that information cause no instance is created yet.

Reply:
And MEF don't tell us the type the metadata is defined on to use reflection to get that information.

Reply:
I am frustrated to have to repeat an info but if we don't have the choice then so be it...

Reply:
I think we could if we go for non attribute based composition (by using the convention based approach) but that would be out of scope for this PR

Reply:
I was thinking of that too, but haven't got the time to do some tries today.
To me, it is a part of this PR since composition is essentially used for commands and arguments so it is justified.
But I could understand if you prefer to finish this PR before starting the next big one.

Reply:
I think we can support both models in the future, without having to make an breaking change for plugin authors. So I think it is out of scope.
Cause if we enforce some kind of convention, people just can remove parts of the attributes, so it's is an additional way (from the technical perspective we are the ones that provide the missing information).

Reply:
Well since we already are breaking (nearly) everything I think it is best to also simplify everything.
If the convention allows us to easily maintain the project and the plugin author to create plugin without having to use a lot of attributes we should go for it.

Reply:
Okay let's address this with a separate issue and put it on the v1.0 list: #353